src/game.rs

```diff
@@ -966,6 +966,117 @@ impl PostFlopGame {
+    }
+
+    fn reverse_push_actions(&self, node: &PostFlopNode, info: &mut BuildTreeInfo) {
+        info.memory_usage_nodes += vec_memory_usage(&node.actions);
+        info.memory_usage_nodes += vec_memory_usage(&node.children);
+        info.num_storage_actions += node.num_elements as u64;
+    }
+
+    fn calculate_removed_line_info_recursive(&self, node: &PostFlopNode, info: &mut BuildTreeInfo) {
+        if node.is_terminal() {
+            return;
+        }
+
+        if node.is_chance() {
+            self.reverse_push_chances(node, info);
+        } else {
+            self.reverse_push_actions(node, info);
+        }
+    }
+
+    /// Remove lines after building the PostFlopGame but before allocating memory.
+    /// This allows the removal of chance-specific lines (e.g., remove overbets on
+    /// board-pairing turns) which we cannot do while building an action tree.
+    pub fn remove_lines(&mut self, lines: &Vec<Vec<Action>>) -> Result<(), String> {
+        if self.state <= State::Uninitialized {
+            return Err("Game is not successfully initialized".to_string());
+        } else if self.state == State::MemoryAllocated {
+            return Err("Game has already been allocated".to_string());
+        }
+
+        for line in lines.iter() {
+            let mut root = self.root();
+            let info = self.remove_line_recursive(&mut root, line)?;
+            self.misc_memory_usage -= info.memory_usage_nodes;
+            self.num_storage_actions -= info.num_storage_actions;
+            self.num_storage_chances -= info.num_storage_chances;
+        }
+        Ok(())
+    }
+
+    /// Recursively remove a line from a PostFlopGame tree
+    fn remove_line_recursive(
+        &self,
+        node: &mut PostFlopNode,
+        line: &[Action],
+    ) -> Result<BuildTreeInfo, String> {
+        if line.is_empty() {
+            return Err("Empty line".to_string());
+        }
+
+        if node.is_terminal() {
+            return Err("Unexpected terminal node".to_string());
+        }
+
+        let action = line[0];
+        let search_result = node.actions.binary_search(&action);
+        if search_result.is_err() {
+            return Err(format!("Action does not exist: {action:?}"));
+        }
+
+        let index = search_result.unwrap();
+        if line.len() > 1 {
+            let result = self.remove_line_recursive(&mut node.children[index].lock(), &line[1..]);
+            if line.len() == 2 {}
+            return result;
+        }
+
+        // Remove action/children at index. To do this we must
+        // 1. compute the storage space required by the tree rooted at action index
+        // 2. remove children and actions
+        // 3. re-define num_elements and num_elements_aux after we remove children and actions
+
+        // STEP 1
+        let mut info = BuildTreeInfo {
+            memory_usage_nodes: mem::size_of::<PostFlopNode>() as u64,
+            num_storage_actions: 0,
```

Comment:
Should be `self.num_private_hands(node.player as usize)` when the last action of the `line` is not a chance action.
```suggestion
            num_storage_actions: self.num_private_hands(node.player as usize),
```